decryptor/postgresql/pg_decryptor.go

```diff
@@ -291,9 +351,15 @@ func checkInlinePoisonRecordInBlock(block []byte, decryptor base.Decryptor, logg
 			if index == utils.NotFound {
 				return nil
 			}
-			currentIndex += index
-			if err := checkWholePoisonRecord(block[currentIndex:], decryptor, logger); err != nil {
-				return err
+			if len(block[index:]) > base.GetMinAcraStructLength() {
```

Comment:
If there is not enough space to fit even a header then why not exit early instead of continuing the loop?